drivers/flash/soc_flash_nrf.c

```diff
@@ -88,16 +88,18 @@ static inline bool is_aligned_32(u32_t data)
 	return (data & 0x3) ? false : true;
 }
 
-static inline bool is_regular_addr_valid(off_t addr, size_t len)
+static inline bool is_regular_addr_valid(off_t *addr, size_t len)
```

Comment:
should this be `check_and_adjust_addr()`?

Reply:
Or maybe `addr_ensure()` is better

Reply:
or `is_regular_range_valid()`

Reply:
I wouldn't correct the address in `is_regular_addr_valid()`. It's easier to read if after the validation is done to say:
`addr += DT_FLASH_BASE_ADDRESS`, I guess the compiler will even eliminate these lines if `DT_FLASH_BASE_ADRESS=0`

Reply:
I agree. I expect predicate functions (is_xyz(foo) is asking a question about foo) not to modify their arguments like this.

Reply:
I resign from shadow offset modification - now this is done directly when needed